src/components/MenuPanel/MenuPanelAppsLoader.js

```diff
@@ -1,78 +1,134 @@
 import React from 'react'
 import styled from 'styled-components'
-import { spring, Motion } from 'react-motion'
-import { spring as springConf } from '@aragon/ui'
+import { Spring, animated } from 'react-spring'
+import { IconAttention, Text } from '@aragon/ui'
 import { lerp } from '../../math-utils'
+import { noop } from '../../utils'
 import LoadingRing from '../LoadingRing'
-
-const SPRING = springConf('fast')
+import {
+  APPS_STATUS_ERROR,
+  APPS_STATUS_READY,
+  APPS_STATUS_LOADING,
+} from '../../symbols'
 
 class MenuPanelAppsLoader extends React.Component {
-  state = { hideLoader: false, loadingTransitionDone: false }
-
-  componentWillReceiveProps({ loading }) {
-    if (loading && !this.props.loading) {
-      this.setState({ hideLoader: false, loadingTransitionDone: false })
-    }
+  static defaultProps = {
+    children: noop,
+    itemsCount: 0,
+    appStatus: APPS_STATUS_LOADING,
+    onRetry: noop,
   }
 
-  handleRest = () => {
-    const { loading } = this.props
+  state = {
+    showApps: false,
+  }
 
-    if (loading) {
-      return
+  componentWillReceiveProps({ appsStatus }) {
+    if (appsStatus !== this.props.appsStatus) {
+      this.setState({ showApps: false })
```

Comment:
On first glance this was quite confusing; it looks like if the `appsStatus` changes, we never show the apps.

If I understand it right, setting this to false will cause the spring to reset and call `handleRest()` once it's done resetting to restart the animation of loading the apps if it is actually ready?

Reply:
> If I understand it right, setting this to false will cause the spring to reset and call handleRest() once it's done resetting to restart the animation of loading the apps if it is actually ready?

Yes, we always want to hide apps if the status changes, and only after the animation we can expand the menu to display them. This is to have the two steps animation.